### ISSUE TITLE
Missing tests for `gen_pauli_x` and `gen_pauli_z`

### ISSUE THREAD

**Issue by purva-thakre** [vprusso/toqito#485]
Related to #478 

Tests for these functions are missing from `toqito/matrices/tests`

| Tests in `matrices/tests`  | Defined functions in `matrices/` |
| ------------- | ------------- |
| ![image](https://github.com/vprusso/toqito/assets/66048318/f74a2639-6271-4d40-ae97-12704641e7b9) | ![image](https://github.com/vprusso/toqito/assets/66048318/63ba0c3a-0815-4a1b-930d-da5031dc4172) |

https://github.com/vprusso/toqito/tree/eb5820721884dae5b9fdaad68913b2d96785167d/toqito/matrices/tests

Interestingly enough, both these functions are not flagged as uncovered by `pytest`. 

![image](https://github.com/vprusso/toqito/assets/66048318/dc332c71-a956-493d-9d4f-956f7c8943c4)

This issue is to find where the tests for these functions are or add files to test them. If it's the latter, need to figure out why `pytest` is not flagging these.

**Comment by purva-thakre** [comment]
@vprusso Looking at the commit history for both functions, I don't think the tests for these functions were added. Wanted to double-check with you as to what happened. Maybe the tests got lost in other modules?

**Comment by purva-thakre** [comment]
Looking through the old commits, I don't see the tests in the previous `tests` directory. 

We changed the directory in #257 and if I look at another PR #255 merged before the restructured directory, the tests don't exist there. 

https://github.com/vprusso/toqito/tree/63c86059f06f994e23be93f3a4626c0b1717b464/tests/test_matrices

Maybe this is also a `pytest` issue as it does not flag the uncovered lines.

**Comment by purva-thakre** [comment]
Interestingly, if I look at the full coverage report, pytest things there are tests for these functions. 

![image](https://github.com/vprusso/toqito/assets/66048318/7d3d39e1-d8bf-4e47-8dfa-ff9961561d82)
![image](https://github.com/vprusso/toqito/assets/66048318/d3392dc3-6efa-4777-ab00-8d9d4133a780)

**Comment by purva-thakre** [comment]
Maybe `pytest` thinks below covers the expected in the another function. 

https://github.com/vprusso/toqito/blob/eb5820721884dae5b9fdaad68913b2d96785167d/toqito/matrices/gen_pauli.py#L74-L75

**Comment by vprusso** [comment]
Hmm, so I think these are covered because there is a test for `gen_pauli` which calls both of these functions. As there isn't any branching in either, testing them via that one function should be all that is required to have them covered from that. 

But yes, having a few specific tests for them explicitly might be a good idea, just to be more thorough and explicit.

**Comment by purva-thakre** [comment]
To close this issue after #478 is merged, in the folder `toqito/matrices/tests`, add a file labeled `test_gen_pauli_x` and another labeled ``test_gen_pauli_z` for testing the functions via `pytest`. 

List of all possible tests:

1. Verify the shape of the matrices is what we expect
2. Verify the expected output is what we expect for a bunch of different values for `dim`. 

It might also be better to use the parameterized option in pytest to consolidate a bunch of identical tests.
https://docs.pytest.org/en/7.3.x/how-to/parametrize.html

Here's an example test in `toqito` where pytest verifies an array output.

https://github.com/vprusso/toqito/blob/eb5820721884dae5b9fdaad68913b2d96785167d/toqito/perms/tests/test_antisymmetric_projection.py#L19-L32
